### PR TITLE
Make sure the tag compound is added back to the itemstack on creation

### PR DESCRIPTION
**Pull request by clienthax**

### CODE REVIEW

File: src/main/java/org/spongepowered/common/item/inventory/SpongeItemStackSnapshot.java [hunk 1]
```diff
@@ -141,7 +141,9 @@ public int getCount() {
 
     @Override
     public ItemStack createStack() {
-        return this.privateStack.copy();
+        net.minecraft.item.ItemStack itemStack = (net.minecraft.item.ItemStack) this.privateStack.copy();
```
Comment (Mumfrey) [inline src/main/java/org/spongepowered/common/item/inventory/SpongeItemStackSnapshot.java]: Ought to be `ItemStackUtil.cloneDefensiveNative(this.privateStack)`

File: src/main/java/org/spongepowered/common/item/inventory/SpongeItemStackSnapshot.java [hunk 2]
```diff
@@ -141,7 +141,9 @@ public int getCount() {
...
+        itemStack.setTagCompound(this.compound);
+        return (ItemStack) itemStack;
```
Comment (Mumfrey) [inline src/main/java/org/spongepowered/common/item/inventory/SpongeItemStackSnapshot.java]: Ought to be `ItemStackUtil.fromNative(itemStack)`

File: src/main/java/org/spongepowered/common/item/inventory/SpongeItemStackSnapshot.java [hunk 3]
```diff
@@ -141,7 +141,9 @@ public int getCount() {
 
     @Override
     public ItemStack createStack() {
-        return this.privateStack.copy();
+        net.minecraft.item.ItemStack itemStack = (net.minecraft.item.ItemStack) this.privateStack.copy();
+        itemStack.setTagCompound(this.compound);
```
Comment (gabizou) [inline src/main/java/org/spongepowered/common/item/inventory/SpongeItemStackSnapshot.java]: If anything, the compound should be copied as well.

File: src/main/java/org/spongepowered/common/item/inventory/SpongeItemStackSnapshot.java [hunk 4]
```diff
@@ -141,7 +142,11 @@ public int getCount() {
 
     @Override
     public ItemStack createStack() {
-        return this.privateStack.copy();
+        net.minecraft.item.ItemStack nativeStack = ItemStackUtil.cloneDefensiveNative(ItemStackUtil.toNative(this.privateStack.copy()));
```
Comment (Aaron1011) [inline src/main/java/org/spongepowered/common/item/inventory/SpongeItemStackSnapshot.java]: I don't think all of this is necessary, as `cloneDefensiveNative` calls `ItemStack#copy`

**Pull request merged**